lib/block_view_dao_coin_limit_order.go

```diff
@@ -1347,13 +1295,29 @@ func (bav *UtxoView) _getDAOCoinLimitOrderEntry(inputEntry *DAOCoinLimitOrderEnt
 
 	// First check if we have the order entry in the UTXO view.
 	outputEntry, _ := bav.DAOCoinLimitOrderMapKeyToDAOCoinLimitOrderEntry[inputEntry.ToMapKey()]
-
 	if outputEntry != nil {
 		return outputEntry, nil
 	}
 
 	// If not, next check if we have the order entry in the database.
-	return bav.GetDbAdapter().GetDAOCoinLimitOrder(inputEntry)
+	var err error
+	outputEntry, err = bav.GetDbAdapter().GetDAOCoinLimitOrder(inputEntry.OrderID)
+	if err != nil {
+		return nil, err
+	}
+	if outputEntry == nil {
+		return nil, nil
+	}
+
+	// Confirm that the existing order entry in the database found by OrderID
+	// was created by this transactor. Otherwise, it would be possible for a
+	// user to delete an order created by another user! In this case, we don't
+	// throw an error but instead just return nil, i.e. no order found.
+	if !inputEntry.TransactorPKID.Eq(outputEntry.TransactorPKID) {
+		return nil, nil
+	}
```

Comment:
I think this check belongs in the connect logic. There may be reasons in the future that we want this function to just return an order entry given an orderId